### ISSUE TITLE
Support Chinese file name?

### ISSUE THREAD

**Issue by codwam** [ZipArchive/ZipArchive#658]
### Steps to reproduce
1. Unzip a zip package containing Chinese file names

### Expected behavior
Chinese file names can be correctly encoded

### Actual behavior
Chinese file name is garbled

### Version of ZipArchive
latest

**Comment by hippietrail** [comment]
Can you include an example?

Seeing exactly how it's garbled might help us to see if it's using one of the non-Unicode Chinese text encodings or is some other problem.

**Comment by Zzy0706** [comment]
try this
```
let zipPath = url.path(percentEncoded: false)
SSZipArchive.createZipFileAtPath(zipPath, withContentsOfDirectory: sampleDataPath)
```

**Comment by Coeur** [comment]
Proper **unicode** support (for compression) should have come with #564, released with ZipArchive 2.2.3 in April 2020:
https://github.com/ZipArchive/ZipArchive/releases/tag/2.2.3

And in a zip format, based on the specs (https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT), there are literally only two possibilities:
- IBM Code Page 437 (which has zero Chinese characters)
- Unicode

So, in the absence of a sample archive or a sample code, all I can say is that Chinese filenames are well supported, based on our unit test:
https://github.com/ZipArchive/ZipArchive/blob/738a6b8bcb08bd107e38d3e7e5bd337632298e28/Example/ObjectiveCExampleTests/SSZipArchiveTests.m#L417

This test is present since at least year 2015 and ZipArchive 0.3.3.
